logic/resolvable/Concludable.java

```diff
@@ -188,8 +188,13 @@ boolean unificationSatisfiable(ThingVariable concludableThingVar, ThingVariable
         }
 
         if (!concludableThingVar.value().isEmpty() && !conclusionThingVar.value().isEmpty()) {
-            // TODO: detect value contradictions between constant predicates
-            satisfiable &= true;
+            assert conclusionThingVar.value().size() == 1;
+            ValueConstraint<?> conclusionValueConstraint = conclusionThingVar.value().stream().findFirst().get();
+
+            assert conclusionValueConstraint.predicate().isEquality() &&
+                    conclusionValueConstraint.predicate().asEquality().equals(EQ);
+
+            satisfiable &=  concludableThingVar.value().stream().allMatch( v -> v.isSatisfiedByValueOf(conclusionValueConstraint) );
```

Comment:
allMatch may be misleading, as different constraints on the same variable may be spread across concludables.